File: packages/sdk-ts/examples/query_agent.ts
import { GeckoVPPClient } from "../src/index.js";

const client = new GeckoVPPClient({
  baseURL: process.env.GECKO_API ?? "http://localhost:8000",
  tenantId: process.env.GECKO_TENANT ?? "11111111-1111-1111-1111-111111111111",
});

const questions: Array<[string, "dispatcher_analyst" | "market_analyst" | "energy_advisor" | "battery_coach"]> = [
  ["що сьогодні з виробництвом?", "dispatcher_analyst"],
  ["який очікуваний дохід на ВДР завтра?", "market_analyst"],
  ["чи варто заряджати батарею зараз?", "battery_coach"],
];

for (const [q, persona] of questions) {
  console.log(`\n[${persona}] ❓ ${q}`);
  const res = await client.agents.query(persona, q);
  console.log(`  → ${res.answer}`);
  console.log(`  intent: ${res.intent}, confidence: ${res.confidence}`);
  if (res.evidence?.length) {
    for (const e of res.evidence) {
      console.log(`  · ${e.label}: ${e.value}`);
    }
  }
}

File: packages/sdk-py/examples/query_agent.py
"""Ask 3 questions to 3 different persona agents."""
import os
from gecko_vpp_sdk import GeckoVPPClient

with GeckoVPPClient(
    base_url=os.environ.get("GECKO_API", "http://localhost:8000"),
    tenant_id=os.environ.get("GECKO_TENANT", "11111111-1111-1111-1111-111111111111"),
) as client:
    questions = [
        ("dispatcher_analyst", "що сьогодні з виробництвом?"),
        ("market_analyst", "який очікуваний дохід на ВДР завтра?"),
        ("battery_coach", "чи варто заряджати батарею зараз?"),
    ]
    for persona, q in questions:
        print(f"\n[{persona}] ❓ {q}")
        res = client.agents_query(persona, q)
        print(f"  → {res['answer']}")
        print(f"  intent: {res['intent']}, confidence: {res['confidence']}")
        for e in res.get("evidence", []):
            print(f"  · {e['label']}: {e['value']}")
